### PR TITLE
Fix for issue #705

### PR DESCRIPTION
**Pull request by tarzasai**
Transmission plugin ignores addpaused for magnet links

### CODE REVIEW

**Comment by tarzasai** [comment]
The test case:
```
tasks:
  testt1:
    disable:
      - seen
      - seen_info_hash
    thetvdb_lookup: yes
    mock:
      - { title: 'The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]', url: 'magnet:?xt=urn:btih:DCC70364B791CB6CEBAADA04C4093C220D8257EE&dn=The+Big+Bang+Theory+S09E11+HDTV+x264-LOL%5Bettv%5D&tr=udp%3A%2F%2Ftracker.glotorrents.com%3A6969%2Fannounce&tr=udp%3A%2F%2Fglotorrents.pw%3A6969%2Fannounce&tr=http%3A%2F%2Ftracker.trackerfix.com%2Fannounce&tr=udp%3A%2F%2Feddie4.nl%3A6969%2Fannounce&tr=udp%3A%2F%2F9.rarbg.to%3A2710%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.aletorrenty.pl%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337%2Fannounce&tr=http%3A%2F%2Ftracker.tfile.me%2Fannounce&tr=http%3A%2F%2Fexplodie.org%3A6969%2Fannounce&tr=udp%3A%2F%2Fp4p.arenabg.com%3A1337&tr=http%3A%2F%2Fbigfoot1942.sektori.org%3A6969%2Fannounce&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80%2Fannounce&tr=udp%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce' }
    all_series:
      tracking: backfill
    accept_all: yes
    transmission:
      host: 192.168.1.200
      port: 9092
      addpaused: yes
      main_file_only: yes
      include_subs: yes
      magnetization_timeout: 90
      path: C:\Storage\Videos\Stage\series
      content_filename: "{{ tvdb_series_name }} - {{ tvdb_season }}x{{ tvdb_episode|pad(2) }}"
```
output:
```
[...]
2016-01-08 21:56 INFO     transmission  testt1          "The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]" torrent added to transmission
2016-01-08 21:56 DEBUG    transmission  testt1          Waiting 90 seconds for "The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]" to magnetize
2016-01-08 21:56 INFO     transmission  testt1          Torrent "The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]" stopped because of addpaused=yes
2016-01-08 21:56 DEBUG    transmission  testt1          Cannot ensure content_filename is unique when adding to a remote transmission daemon.
2016-01-08 21:56 DEBUG    transmission  testt1          File The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]/the.big.bang.theory.911.hdtv-lol[ettv].mp4 renamed to The Big Bang Theory - 9x11.mp4
2016-01-08 21:56 DEBUG    transmission  testt1          Downloading 1 of 2 files in torrent.
[...]
```

File: flexget/plugins/plugin_transmission.py [hunk 1]
```diff
@@ -473,6 +473,9 @@ def _wait_for_files(cli, r, timeout):
  file list unavailable for processing.' % entry['title'])
                             else:
                                 total_size = cli.get_torrent(r.id, ['id', 'totalSize']).totalSize
+                                if 'paused' in options['post'] and options['post']['paused'] == True:
+                                    cli.stop_torrent(r.id)
+                                    log.info('Torrent "%s" stopped because of addpaused=yes' % entry['title'])
```
Comment (gazpachoking) [inline flexget/plugins/plugin_transmission.py]: Maybe this should be at verbose or debug level?

**Comment by gazpachoking** [comment]
Oh hmm. This fix is inside the 
```
                if ('main_file_only' in options['post'] and options['post']['main_file_only'] == True or 
                   'content_filename' in options['post'] or skip_files):
```
if statement, and won't take effect if that statement isn't true. Wouldn't right after here be a more appropriate place? https://github.com/tarzasai/Flexget/blob/addpaused_transmission_magnets/flexget/plugins/plugin_transmission.py#L594

**Comment by tarzasai** [comment]
@gazpachoking you're right, it was the wrong place. but there is a bad news: when the metadata are downloaded instantly, a subsequent call to cli.stop_torrent() has no effect - it can't stop the torrent, so for now I've added a sleep. if anyone has a better solution...

**Comment by gazpachoking** [comment]
Is the issue when content_filename or main_file_only isn't set? (and thus, it doesn't wait for the metadata to be fetched)
If so, I'd say we could do something like `if not downloaded: _wait_for_files()` before calling the stop.

**Comment by tarzasai** [comment]
well, let's see:
* with content_filename only it downloads the metadata and stop.
* with main_file_only only or both it downloads the metadata but keep going.

_wait_for_files() helps a bit: we get the metadata and stop the torrent, but content_filename cannot rename the file - i've just added a `if not downloaded: _wait_for_files(cli, r, options['post'].get('magnetization_timeout', 90))` before the cli call btw.

**Comment by liiight** [comment]
@tarzasai is this ready?

**Comment by tarzasai** [comment]
I think it is. I can't find how to solve it without the delay, but at least it works now.

**Comment by cvium** [comment]
Should probably update the wiki.

**Comment by tarzasai** [comment]
I wrote a note in the advanced section - and it probably needs a review.

**Pull request merged**